tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/contentassist/TclTemplateProposalProvider.xtend

```diff
@@ -56,76 +60,221 @@ class TclTemplateProposalProvider extends DefaultTemplateProposalProvider {
 		super.createTemplates(templateContext, context, acceptor)
 
 		val model = context.currentModel
+		if (model instanceof ComponentTestStepContext) {
+			model?.proposeAvailableInteractions(templateContext, context, acceptor)
+		}
+		if (model instanceof MacroTestStepContext) {
+			model?.proposeAvailableInteractions(templateContext, context, acceptor)
+		}
 		if (model instanceof TestStep) {
 			if (model.hasComponentContext) {
 				val componentContext = model.componentContext
-				componentContext?.proposeAvailableInteractions(model, templateContext, context, acceptor)
-			}else if (model.hasMacroContext){
+				componentContext?.proposeAvailableInteractions(templateContext, context, acceptor)
+			} else if (model.hasMacroContext) {
 				val macro = model.macroContext
-				macro?.proposeAvailableInteractions(model, templateContext, context, acceptor)
+				macro?.proposeAvailableInteractions(templateContext, context, acceptor)
 			}
 		}
 	}
+	
+	/**
+	 * give this' node grammarElement as 'clazz'
+	 * <pre>
+	 * iff: the node is a leaf node (thus not null)
+	 *      the grammarElement is of the requested type
+	 * else: null
+	 * </pre>
+	 */
+	private def <T> getNodeAsGrammarElement(INode node, Class<T> clazz) {
+		if ((node !== null) && (node instanceof LeafNode)) {
+			val grammarElement = (node as LeafNode).grammarElement
+			if (clazz.isAssignableFrom(grammarElement.class)) {
+				return grammarElement as T
+			}
+		} 
+		return null
+	}
+	
+	private def boolean isRightBehindDash(ContentAssistContext context) {
+		val lastCompleteKeyword = context.lastCompleteNode.getNodeAsGrammarElement(Keyword)
+		return lastCompleteKeyword !== null && lastCompleteKeyword.value == '-'
+	}
+
+	private def boolean isRightBehindAssignment(ContentAssistContext context) {
+		val lastCompleteKeyword = context.lastCompleteNode.getNodeAsGrammarElement(Keyword)
+		return lastCompleteKeyword !== null && lastCompleteKeyword.value == '='
+	}
+
+	private def boolean rightBehindSpace(ContentAssistContext context) {
+		val peekedString = context.peekBehindContexCursor(1)
+		return peekedString == ' '
+	}
+	
+	private def String peekBehindContexCursor(ContentAssistContext context, int length) {
```

Comment:
You're actually peeking **ahead**, right? "Behind" might be confusing.

Reply:
hope the comment (later in the source code) clarifies a bit that the peek goes back/behind the cursor position